.github/workflows/pg_upgrade-test.yaml

```diff
@@ -0,0 +1,117 @@
+name: pg_upgrade test
+"on":
+  push:
+    branches:
+      - main
+      - prerelease_test
+  pull_request:
+
+jobs:
+  pg_upgrade_test:
+    name: pg_upgrade test from PG${{ matrix.pg_version_old }} to PG${{ matrix.pg_version_new }}
+    runs-on: 'ubuntu-latest'
+    strategy:
+      matrix:
+        include:
+          - pg_version_old: 14 # 14 to 15
+            pg_version_new: 15
+          - pg_version_old: 14 # 14 to 16
+            pg_version_new: 16
+          - pg_version_old: 15 # 15 to 16
+            pg_version_new: 16
+      fail-fast: false
+    env:
+      OUTPUT_DIR: ${{ github.workspace }}/pg_upgrade_test
+
+    steps:
+      - name: Install Linux Dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install pip postgresql-common libkrb5-dev
+
+      # Use custom testgres repo until upstream PR don't get merged and released
+      # https://github.com/postgrespro/testgres/pull/125
+      - name: Checkout testgres
+        uses: actions/checkout@v4
+        with:
+          repository: 'fabriziomello/testgres'
```

Comment:
They merged the PR so I've changed this PR to get the code from upstream. When they officially release new version we can change it and use the pip package instead of building direct from source code.